### PR TITLE
feat: switch to klog v2 and structured logging

### LINKED ISSUE #207: (feat): structured log
Would be nice to provide arg option for structured log format ( json )

### LINKED ISSUE #245: Update to go 1.15
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Switch to using `klog/v2`
- Enable structured logging
- Bump dependencies and go version to `1.15`
- Reorganize imports
```
import(
  <standard libs>

  <internal libs>

  <external libs>
)
```

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #245 
fixes #207 

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Special Notes for Reviewers**:

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml [hunk 1]
```diff
@@ -26,6 +26,9 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
             - --endpoint=unix://C:\\provider\\azure.sock
+            {{- if .Values.logFormatJSON }}
```
Comment (ritazh) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:29]: Do we also want to add:
```
            {{- if .Values.logVerbosity }}
            - -v={{ .Values.logVerbosity }}
            {{- end }}
```

Comment (aramase) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:29]: thank you! Added it to the helm charts and finished rebase.

**Review by ritazh — APPROVED** [review]
lgtm

**Comment by aramase** [comment]
```bash
I1106 01:07:19.312951       1 main.go:77] Listening for connections on address: //provider/azure.sock
I1106 01:07:28.739053       1 provider.go:239] "unmarshaled key vault objects" keyVaultObjects=[{ObjectName:secret1 ObjectAlias: ObjectVersion: ObjectType:secret ObjectFormat: ObjectEncoding:} {ObjectName:secret2 ObjectAlias: ObjectVersion: ObjectType:secret ObjectFormat: ObjectEncoding:} {ObjectName:ingress-tls-pfx ObjectAlias: ObjectVersion: ObjectType:secret ObjectFormat:pfx ObjectEncoding:}] count=3 pod="default/nginx-66b6676558-79qt5"
I1106 01:07:28.739155       1 provider.go:250] "fetching object from key vault" objectName="secret1" objectType="secret" keyvault="kindkv" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:28.742429       1 auth.go:148] "using service principal to retrieve access token" clientID="c68a##### REDACTED #####e807" secret="058c##### REDACTED #####89f6" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.034496       1 provider.go:283] "successfully wrote file" file="secret1" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.034555       1 provider.go:250] "fetching object from key vault" objectName="secret2" objectType="secret" keyvault="kindkv" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.034736       1 auth.go:148] "using service principal to retrieve access token" clientID="c68a##### REDACTED #####e807" secret="058c##### REDACTED #####89f6" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.550613       1 provider.go:283] "successfully wrote file" file="secret2" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.550677       1 provider.go:250] "fetching object from key vault" objectName="ingress-tls-pfx" objectType="secret" keyvault="kindkv" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:33.553670       1 auth.go:148] "using service principal to retrieve access token" clientID="c68a##### REDACTED #####e807" secret="058c##### REDACTED #####89f6" pod="default/nginx-66b6676558-79qt5"
I1106 01:07:34.300845       1 provider.go:283] "successfully wrote file" file="ingress-tls-pfx" pod="default/nginx-66b6676558-79qt5"
```

**Pull request merged**